### PR TITLE
Update jda

### PR DESCRIPTION
**Pull request by SnakeJ419**

### CODE REVIEW

File: src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java [hunk 1]
```diff
@@ -5,21 +5,35 @@
...
+import net.dv8tion.jda.api.exceptions.ErrorResponseException;
 import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.requests.ErrorResponse;
 import org.jetbrains.annotations.NotNull;
 
 public class ReactionProcessing extends ListenerAdapter {
+    @Override
+    public void onMessageDelete(@NotNull MessageDeleteEvent event) {
+
+    }
```
Comment (SnakeJ419) [inline src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:22]: why is this here if we're doing nothing with it?

File: src/main/java/com/wylxbot/wylx/Core/Util/Helper.java [hunk 2]
```diff
@@ -59,11 +63,11 @@ private ValidateContainer(Runnable runnable, long messageID, long userID){
         @Override
         public void onGenericMessageReaction(@NotNull GenericMessageReactionEvent event) {
             if(event.getMessageIdLong() == messageID && event.getUserIdLong() == userID){
-                if(event.getReactionEmote().getAsCodepoints().equals(CHECK)){
+                if(event.getEmoji().equals(CHECK)){
                     event.getChannel().deleteMessageById(messageID).queue();
                     runnable.run();
                     Wylx.getInstance().getJDA().removeEventListener(this);
-                } else if(event.getReactionEmote().getAsCodepoints().equals(X)){
+                } else if(event.getEmoji().equals(X)){
```
Comment (SnakeJ419) [inline src/main/java/com/wylxbot/wylx/Core/Util/Helper.java:70]: have you tested that using .equals on the `EmojiUnion` class comes out as we would expect?

File: src/main/java/com/wylxbot/wylx/Core/Util/Helper.java [hunk 3]
```diff
@@ -161,8 +137,18 @@ public void run() {
...
+            Emoji.fromUnicode("U+31U+fe0fU+20e3"),
+            Emoji.fromUnicode("U+32U+fe0fU+20e3"),
+            Emoji.fromUnicode("U+33U+fe0fU+20e3"),
+            Emoji.fromUnicode("U+34U+fe0fU+20e3"),
+            Emoji.fromUnicode("U+35U+fe0fU+20e3"),
+            Emoji.fromUnicode("U+36U+fe0fU+20e3"),
+            Emoji.fromUnicode("U+37U+fe0fU+20e3"),
+            Emoji.fromUnicode("U+38U+fe0fU+20e3"),
+            Emoji.fromUnicode("U+39U+fe0fU+20e3")
+            );
```
Comment (SnakeJ419) [inline src/main/java/com/wylxbot/wylx/Core/Util/Helper.java:151]: Is there a way to get emoji from the form of `:one:` now? If not could we commend these lines since youve spaced them out so nicely already

File: src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/RoleMenu.java [hunk 4]
```diff
@@ -46,28 +49,38 @@ public RoleMenu(String title, String messageID, String channelID, String guildID
...
-                if (emote == null) {
-                    return;
-                } else {
-                    emoji = Emoji.fromEmote(emote);
+            EmojiUnion emoji;
+            if (emojiInput.matches("[0-9]+")) {
+                RichCustomEmoji customEmoji = jda.getEmojiById(emojiInput);
+                if (customEmoji == null) {
+                    updateMenu = true;
+                    continue;
```
Comment (SnakeJ419) [inline src/main/java/com/wylxbot/wylx/Commands/Roles/RolesUtil/RoleMenu.java:70]: have you done any tests to make sure it does actually work with custom emoji still + that it doesnt break the data base, I can setup a role menu with custom emotes w/ my dev bot on main then restart it with the new jda to test the role menu and see if youd like

**Review by SnakeJ419 — COMMENTED** [review]
just a couple of small things, mostly just things i dont know the answer to not things that seem actually problematic

File: src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java (hunk 1 above)
Comment (1Revenger1) [reply src/main/java/com/wylxbot/wylx/Core/Processing/ReactionProcessing.java:22]: Oh, me probably trying to figure out if I could handle role menus being removed by the message being deleted.

File: src/main/java/com/wylxbot/wylx/Core/Util/Helper.java (hunk 2 above)
Comment (1Revenger1) [reply src/main/java/com/wylxbot/wylx/Core/Util/Helper.java:70]: What uses this? I'm not exactly sure which commands have the validation checks.

File: src/main/java/com/wylxbot/wylx/Core/Util/Helper.java (hunk 3 above)
Comment (1Revenger1) [reply src/main/java/com/wylxbot/wylx/Core/Util/Helper.java:151]: Not for Unicode Emojis afaik, unless I missed it. Don't think so though

File: src/main/java/com/wylxbot/wylx/Core/Util/Helper.java (hunk 2 above)
Comment (SnakeJ419) [reply src/main/java/com/wylxbot/wylx/Core/Util/Helper.java:70]: this is for the big moderation things like clearing over 15 messages

File: src/main/java/com/wylxbot/wylx/Core/Util/Helper.java (hunk 3 above)
Comment (SnakeJ419) [reply src/main/java/com/wylxbot/wylx/Core/Util/Helper.java:151]: sadge

File: src/main/java/com/wylxbot/wylx/Core/Util/Helper.java (hunk 2 above)
Comment (SnakeJ419) [reply src/main/java/com/wylxbot/wylx/Core/Util/Helper.java:70]: alright i have tested it and it works

**Review by 1Revenger1 — APPROVED** [review]
